department-of-veterans-affairs/va.gov-cms
# VACMS-2896 System Service migration for VISN 2,6,16,21, 23

## Description

See #2896 

## Testing done
Ran the migration.
Looked for migration messages. (messages would be created if any of the entity lookups fail.
Examined select nodes for data.

## Screenshots
Before:
![image](https://user-images.githubusercontent.com/5752113/99455753-4931d300-28f6-11eb-9658-20de98a33a70.png)

After:
![image](https://user-images.githubusercontent.com/5752113/99455775-4f27b400-28f6-11eb-8dbd-2c5fbd6b2737.png)



## QA steps

As an admin, go here `http://pr3368.ci.cms.va.gov/admin/structure/migrate/manage/vamc/migrations`
Examine the migration line for `regional_health_care_service_region_keyed`
![image](https://user-images.githubusercontent.com/5752113/99460237-82ba0c80-28fd-11eb-856a-b374ceca8a10.png)
- [x] validate it shows 1381 unprocessed entries.

Press the execute button on that line to run the migration.
Batch process loader should appear and will run for a while. 
When complete, 
![image](https://user-images.githubusercontent.com/5752113/99461627-21e00380-2900-11eb-908d-1a333b0a419d.png)

- [x] Validate that there are 0 unprocessed nodes.
- [x] Validate that there are 0 messages
- [x] Validate that the processed and imported counts match (1695)

Function of this migration has been previously validated but spot check content by visiting
http://pr3368.ci.cms.va.gov/admin/content?title=&type=regional_health_care_service_des&moderation_state=All&owner=All
- [x] validate that titles look complete with `<service> at <system>` 
example: Advice nurse at VA Southeast Louisiana health care
Visit http://pr3368.ci.cms.va.gov/shreveport-health-care/health-services/telehealth-at-va-shreveport-health-care
- [x] validate VAMC system description exists with three bullets and a link.
- [x] validate it is unpublished
- [x] validate CMS migrator is the author / last revision.
- [x] validate "Posted To" is connected to "VA Shreveport health care"

Visit http://pr3368.ci.cms.va.gov/southeast-louisiana-health-care/health-services/homeless-veteran-care-at-va-southeast-louisiana
- [x] validate VAMC system description exists with 6 bullets.
- [x] validate Posted to  is connected to "VA Southeast Louisiana health care"


Visit http://pr3368.ci.cms.va.gov/st-cloud-health-care/health-services/wound-care-and-ostomy-at-va-st-cloud-health-care
- [x] validate VAMC system description exists with 2 bullets.
- [x] validate Posted to  is connected to "VA St Cloud health care"

## Definition of Done
- [x] All system services from the VISNs 2, 6, 16, 21 and 23 are migrated on prod.

## Post deploy steps
As an admin, go here on prod  `https://prod.cms.va.gov/admin/structure/migrate/manage/vamc/migrations`
Examine the migration line for `regional_health_care_service_region_keyed` and execute it.
It should take a while to run.  When it is complete it should have matching counts for processed and imported and no messages.

### CMS user-facing release note

Is an announcement needed to let editors know of this change?
